kythe/cxx/indexer/cxx/GoogleFlagsLibrarySupport.cc

```diff
@@ -172,28 +174,55 @@ static GraphObserver::NodeId NodeIdForFlag(GraphObserver& Observer,
                              "google/gflag#" + VarId.getRawIdentity());
 }
 
+static std::optional<GraphObserver::NodeId> NodeIdForFlagName(
+    GraphObserver& observer, const GraphObserver::NodeId& VarId,
+    const std::string_view flag_name) {
+  if (const KytheClaimToken* token =
+          clang::dyn_cast<KytheClaimToken>(VarId.getToken())) {
+    proto::VName vname;
+    vname.set_signature(flag_name);
+    vname.set_corpus(token->vname().corpus());
+    vname.set_language("flag");
+    return observer.MintNodeIdForVName(vname);
+  }
+  // Cannot determine the corpus.
+  return std::nullopt;
+}
+
+static void RecordFlagName(GraphObserver& Observer,
+                           const GraphObserver::NodeId& FlagNodeId,
+                           const clang::VarDecl* Decl,
+                           clang::SourceRange Range) {
+  const auto& Context = Decl->getASTContext();
+  const auto& SM = Context.getSourceManager();
+  const char* begin = SM.getCharacterData(Range.getBegin());
+  const char* end = SM.getCharacterData(Range.getEnd());
+  if (begin >= end) {
+    return;
+  }
+  llvm::StringRef FlagName(begin, end - begin);
+  if (auto FlagNameId =
+          NodeIdForFlagName(Observer, FlagNodeId, FlagName.str())) {
```

Comment:
Avoid the unnecessary copy here by having `NodeIdForFlagName` accept `llvm::StringRef` or by using `NodeIdForFlagName(..., {FlagName.data(), FlagName.size()})`